examples/sdk-upload/README.md

```diff
@@ -0,0 +1,334 @@
+# SDK Upload to S3 - A Complete Guide
+
+We are going to implement a simple solution for uploading images to an S3 bucket
+using the AWS SDK.
+
+![upload example](https://cloud.githubusercontent.com/assets/12450298/18955390/2996f122-864f-11e6-92f5-f2b4c631b2d7.png)
+
+### Contents
+- [Creating an S3 Bucket](#step-1---creating-the-bucket)
+- [Creating IAM User with S3 Permissions](#step-2---creating-an-iam-user-with-s3-permissions)
+- [AWS SDK Configuration](#step-3---configure-the-aws-adk)
+- [Implement the SDK](#step-4---implement-the-sdk)
+- [Create a Server](#step-5---create-a-server-to-handle-our-upload-file)
+- [Client Side Code](#step-6---write-the-client-side-code-to-send-our-file-to-the-server)
+- [Take it for a Spin](#take-it-for-a-spin)
+- [Learning Resources](#learning-resources)
+
+### Step 1 - Creating the bucket
+
+In our direct-upload example we walk through the steps of how to create a new
+bucket in S3. Check out [the tutorial](https://github.com/dwyl/image-uploads/blob/master/examples/direct-upload/README.md#step-1---creating-the-bucket)
+and follow the steps.
+
+### Step 2 - Creating an IAM user with S3 permissions
+
+Just as in Step 1 we've already created a step-by-step process for creating a
+new IAM user that has the correct permissions to access the bucket you created
+[above](#step-1---creating-the-bucket). Take a look at [the tutorial](https://github.com/dwyl/image-uploads/blob/master/examples/direct-upload/README.md#step-2---creating-an-iam-user-with-s3-permissions)
+and again follow the steps.
+
+### Step 3 - Configure the AWS SDK
+
+In order to upload to our S3 bucket, we have to use the AWS SDK. Install it using
+the following command:
+
+`$ npm install aws-sdk --save`
+
+Next we'll need to configure our AWS SDK. You can do this in a number of ways.
+We're going to create a credentials file at `~/.aws/credentials`. To do this we
+take the following steps:
+
+* `$ cd` this takes us back to the root of our file system
```

Comment:
I'd recommend people open a ***new terminal window*** to perform these actions so they can close it at the end and go back to their project directory in the _original_ terminal. ...